src/ty/path.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2019
+//     by  Centrality Investments Ltd.
+//     and Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::tm_std::*;
+
+use crate::{
+	form::{CompactForm, Form, MetaForm},
+	utils::is_rust_identifier,
+	IntoCompact, MetaType, Registry,
+};
+use serde::Serialize;
+
+/// Represents a path to a type, represented by the namespace, name and optional
+/// type params.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Debug)]
+#[serde(bound = "F::TypeId: Serialize")]
+pub struct Path<F: Form = MetaForm> {
+	/// The name of the type.
+	name: F::String,
+	/// The namespace in which the type has been defined.
+	namespace: Namespace<F>,
+	/// The generic type parameters of the type in use.
+	#[serde(rename = "params")]
+	type_params: Vec<F::TypeId>,
+}
```

Comment:
Ah okay I see. Maybe an abstraction that contains all of these 3 things is useful.

Reply:
What do you think about the name though? I called it originally `Id` but that is overloaded?. So I just looked at `syn` and took that name: https://docs.rs/syn/1.0.16/syn/struct.Path.html

Reply:
Name is fine imo. :)

Reply:
Also I have chosen to flatten it in the serialization. What do you think about not flattening and have those three fields within a `path` object?

Reply:
How does flattening of those 3 fields mean in serialization? I mean how does it look like?

Reply:
"composite": {
	"name": 8, // RecursiveStruct
	"namespace": [2], // json
	"params": [],
	"fields": [
		{
			"name": 9, // rec
			"type": 8, // Vec<RecursiveStruct>
		}
	]
},